### PR TITLE
After first round

### PR DESCRIPTION
**Pull request by iherman**
I have updated the document to include all various options that we discussed; the goal was to make some sort of a synopsis of where we are. This may help in our discussions going forward.

Cc @aphillips, @r12a : is it o.k. to merge this? (We should not have long discussion on this PR issue, they should happen as "real" issues, but I may have misunderstood/misinterpreted somethings...

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-dir-literal/pull/8.html" title="Last updated on Jun 4, 2019, 12:59 PM UTC (e8704f8)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-dir-literal/8/d124a5b...e8704f8.html" title="Last updated on Jun 4, 2019, 12:59 PM UTC (e8704f8)">Diff</a>

### CODE REVIEW

**Comment by iherman** [comment]
Readable/formatted version: https://raw.githack.com/w3c/rdf-dir-literal/after-first-round/index.html

File: index.html [hunk 1]
```diff
@@ -1,414 +1,482 @@
...
+                <p>
+                    Finally, just as for <a href=#extending-bcp></a>, the interplay between current data deployment and the assignment of such extended language tags may not be obvious.
+                </p>
+            </section>
+        </section>
+        <section id=script-subtag>
+            <h2>Rely on current BCP47 only</h2>
+
+            <p>
+				A different approach stems from the observation that the current BCP-47 can be used, unchanged, to express the necessary base direction using, if necessary, the “script” subtag. For example, the tag <code>he-Hebr</code> reinforces, through the presence of the <code>Hebr</code> subtag, that the language is not only Hebrew, but the string uses the Hebrew script, i.e., it <em>is</em> right-to-left. In other words:
```
Comment (r12a) [inline index.html]: A better example here would be `az-Arab`, since `he` on its own should be sufficient for Hebrew, and it's best not to encourage `he-Hebr` usage.

File: index.html [hunk 2]
```diff
@@ -1,414 +1,482 @@
...
+			</p>
 
-				<pre class=nohighlight>
-					[] ex:example "<bdo dir="ltr">&#x05E4;&#x05E2;&#x05D9;&#x05DC;&#x05D5;&#x05EA; &#x05D4;&#x05D1;&#x05D9;&#x05E0;&#x05D0;&#x05D5;&#x05DD;, W3C</bdo>@he-d-rtl .
+            <pre class=nohighlight>
+					[] ex:example "<bdo dir="ltr">&#x05E4;&#x05E2;&#x05D9;&#x05DC;&#x05D5;&#x05EA; &#x05D4;&#x05D1;&#x05D9;&#x05E0;&#x05D0;&#x05D5;&#x05DD;, W3C</bdo>"@he-Hebr .
```
Comment (r12a) [inline index.html]: Again, he-Hebr -> he, or use Azeri.

File: index.html [hunk 3]
```diff
@@ -1,414 +1,482 @@
...
-				<p>in Turtle, and</p>
+            <p>or</p>
 
-				<pre class=nohighlight>
+            <pre class=nohighlight>
 					"ex:example" : {
-						"@value" : "<bdo dir="ltr">&#x05E4;&#x05E2;&#x05D9;&#x05DC;&#x05D5;&#x05EA; &#x05D4;&#x05D1;&#x05D9;&#x05E0;&#x05D0;&#x05D5;&#x05DD;, W3C</bdo>"
-						"@language" : "he-d-rtl"
+					    "@value" : "<bdo dir="ltr">&#x05E4;&#x05E2;&#x05D9;&#x05DC;&#x05D5;&#x05EA; &#x05D4;&#x05D1;&#x05D9;&#x05E0;&#x05D0;&#x05D5;&#x05DD;, W3C</bdo>",
+					    "@language" : "e-Hebr"
```
Comment (r12a) [inline index.html]: Same again, except that there's an additional typo. 'e'

File: index.html (hunk 2 above)
Comment (iherman) [reply index.html]: Can you give me an azeri example?

Comment (r12a) [reply index.html]: that's the `az-Arab` example in the earlier comment.  See https://r12a.github.io/app-subtags/?lookup=az-Arab

File: index.html (hunk 1 above)
Comment (iherman) [reply index.html]: But... isn't the point of "Rely on BCP-47" the fact that a, possibly redundant, information can be used as an extra info reinforcing the rtl nature? That is what I try to say in:

> For example, the tag he-Hebr reinforces, through the presence of the Hebr subtag, that the language is not only Hebrew, but the string uses the Hebrew script, i.e., it is right-to-left

If what you say is true, i.e., the "he" tag is enough to remove all ambiguities of the Bidi problem (ie, the fact that the string may with a Latin character) then I still do not understand why we need the directional information in the first place... Are all our examples wrong (provided the language tag is used)?

Comment (iherman) [reply index.html]: To think a bit further: _if_ I understand what you seem to suggest in your comment, isn't it correct to say that _the base direction information is only necessary in the absence of a correct language tag_ (by 'correct' I mean including the Script tag in case of ambiguities like Azeri). If that is indeed the case then I really do not see why this information is really necessary for applications like JSON/RDF, because that extra tag carries, in 90% of the cases, redundant information...

Comment (dlongley) [reply index.html]: > To think a bit further: if I understand what you seem to suggest in your comment, isn't it correct to say that the base direction information is only necessary in the absence of a correct language tag (by 'correct' I mean including the Script tag in case of ambiguities like Azeri). If that is indeed the case then I really do not see why this information is really necessary for applications like JSON/RDF, because that extra tag carries, in 90% of the cases, redundant information...

That's what I was thinking... and to be honest, it seems like the problems we've been discussing have arisen *because* a feature has been added to express direction separately from language in some syntaxes. If that had never happened, there would be no reason to worry about what happens when a direction is set (or set "incorrectly") without a language. It seems that the answer is to just specify the proper language if you want your data to be understood correctly.

In other words, if it makes no sense to specify a direction that contradicts the language then the two never should have been decoupled. If there are "non-language" strings that need a special direction to understand them, then, again, no special "direction" feature should have been created for this, rather, it should have used a datatype (or something equivalent). The hack, at least to me, seems to be the notion of "direction" as its own property.

Comment (iherman) [reply index.html]: @dlongley, I think the problem is a bit more nuanced than that...

Consider the acronym "W3C". Making it clear that the direction is ltr is important. However, there is no natural language that MUST be assigned to this acronym: indeed, both "W3C"@en and "W3C"@fr are acceptable and the difference between the two is important because it may decide how a text-to-speech engine would utter the acronym. Ie, direction and language are not 100% bound. Even "W3C"@ar should be acceptable for the very same reason but, in this case, we _do_ need to give a direction somehow. The question is whether "W3C'@ar-Latn is acceptable for this case or not. If it _is_, then it signals the left-to-right typesetting following the proposal in this section.

**Comment by iherman** [comment]
@r12a I have modified the 'pure bcp47' section after our discussions; thanks also for the comment of @macchiati. (I have incorporated the processing steps into the document, too.)

I intend to merge this tonight (my time) to have an updated document for our discussion of tomorrow.

File: index.html [hunk 4]
```diff
@@ -429,6 +443,10 @@ <h2>Rely on the UNICODE formatting characters</h2>
                 The base direction of a string may also be controlled by the Unicode formatting characters U+200E LEFT-TO-RIGHT MARK or U+200F RIGHT-TO-LEFT MARK. This means that user agents receiving such a string could identify the base direction of any given natural language value by scanning the text for the first strong directional character that may include these formatting characters; no further information is strictly necessary. (See, e.g., the <a href="https://w3c.github.io/wpub/#app-bidi-examples">Web Publication draft</a>&nbsp;[[wpub]] for some examples.)
             </p>
 
+            <p>
+                Note that this approach may be part of the previous approach, insofar as the usage of the formatting characters are integral part of the standard BIDI algorithm used in the possible processing step described in <a href="cript-subtag"></a>
```
Comment (r12a) [inline index.html:447]: typo: "cript-subtag"

Comment (iherman) [reply index.html:447]: Oops... done. Thanks.
